src/main/java/me/minidigger/hangar/security/HangarAuthentication.java

```diff
@@ -11,21 +12,30 @@
     private static final long serialVersionUID = 3847099128940870714L;
 
     private final String username;
+    private final Long userId;
     private UsersTable table; // todo replace me
 
     public HangarAuthentication(String username) {
         super(null);
         this.username = username;
+        this.userId = null;
         setAuthenticated(false);
     }
 
-    public HangarAuthentication(String username, UsersTable usersTable, Collection<GrantedAuthority> authorities) {
+    public HangarAuthentication(Collection<? extends GrantedAuthority> authorities, String username, long userId, UsersTable table) {
         super(authorities);
         this.username = username;
-        this.table = usersTable;
-        super.setAuthenticated(true); // must use super, as we override
```

Comment:
and reason you don't set this to authenticated? does spring even accept it if its not true?

Reply:
actually, the auth manager does this, right? I added the auth manager after I wrote this code, so it should be fine

Reply:
actually, I'm not sure... other implementations of AbstractAuthenticationToken do call super.setAuthenticated in their constructors... I wasn't seeing any issues, but I'll add it back.